### PR TITLE
feat(build): #990 move cache directory

### PR DESCRIPTION
**Pull request by sebas031811**

### CODE REVIEW

**Comment by dsalaza4** [comment]
Hi @sebas031811,

Thank you so much for your contribution!

I see other files that also reference the `.makes` folder:

https://github.com/fluidattacks/makes/blob/3004ae797b2ed44806cb9d923d0c4b293dd93c37/src/args/agnostic.nix#L5-L6

https://github.com/fluidattacks/makes/blob/3004ae797b2ed44806cb9d923d0c4b293dd93c37/src/evaluator/modules/default.nix#L62

**Comment by sebas031811** [comment]
Hi @dsalaza4,  
I understood it was just '.makes/cache'. Anyway, there is also a file 'demo.svg', how could I change it?

**Comment by dsalaza4** [comment]
@sebas031811 Everything should be under `.cache/makes` now. 

Regarding the demo, you can leave it the way it is.

File: src/args/agnostic.nix [hunk 1]
```diff
@@ -2,8 +2,8 @@
 # and therefore are agnostic to the framework.
 {
   stateDirs ? {
-    global = "$HOME_IMPURE/.makes/state";
-    project = "$HOME_IMPURE/.makes/state/__default__";
+    global = "$HOME_IMPURE/.cache/state";
+    project = "$HOME_IMPURE/.cache/state/__default__";
```
Comment (dsalaza4) [inline src/args/agnostic.nix:6]:
```suggestion
    global = "$HOME_IMPURE/.cache/makes/state";
    project = "$HOME_IMPURE/.cache/makes/state/__default__";
```

File: docs/src/api/extensions/fundamentals.md [hunk 2]
```diff
@@ -584,8 +584,8 @@ Example:
 
         [DEBUG] Version is 1.0
         [INFO] pwd is /data/github/fluidattacks/makes
-        [INFO] Running tree command on /home/user/.makes/state/example
-        /home/user/.makes/state/example
+        [INFO] Running tree command on /home/user/.cache/state/example
+        /home/user/.cache/state/example
```
Comment (dsalaza4) [inline docs/src/api/extensions/fundamentals.md:588]:
```suggestion
        /home/user/.cache/makes/state/example
```

File: docs/src/api/extensions/fundamentals.md [hunk 3]
```diff
@@ -584,8 +584,8 @@ Example:
 
         [DEBUG] Version is 1.0
         [INFO] pwd is /data/github/fluidattacks/makes
-        [INFO] Running tree command on /home/user/.makes/state/example
-        /home/user/.makes/state/example
+        [INFO] Running tree command on /home/user/.cache/state/example
```
Comment (dsalaza4) [inline docs/src/api/extensions/fundamentals.md:587]:
```suggestion
        [INFO] Running tree command on /home/user/.cache/makes/state/example
```

File: src/cli/main/cli.py [hunk 4]
```diff
@@ -74,9 +74,9 @@
     highlight=False,
     file=io.TextIOWrapper(sys.stderr.buffer, write_through=True),
 )
-MAKES_DIR: str = join(environ["HOME_IMPURE"], ".makes")
-makedirs(join(MAKES_DIR, "cache"), exist_ok=True)
-SOURCES_CACHE: str = join(MAKES_DIR, "cache", "sources")
+MAKES_DIR: str = join(environ["HOME_IMPURE"], ".cache")
```
Comment (dsalaza4) [inline src/cli/main/cli.py:77]:
```suggestion
MAKES_DIR: str = join(environ["HOME_IMPURE"], ".cache/makes")
```

File: src/cli/main/cli.py [hunk 5]
```diff
@@ -74,9 +74,9 @@
...
+makedirs(join(MAKES_DIR, "makes"), exist_ok=True)
```
Comment (dsalaza4) [inline src/cli/main/cli.py:78]:
```suggestion
makedirs(MAKES_DIR), exist_ok=True)
```

File: src/cli/main/cli.py [hunk 6]
```diff
@@ -74,9 +74,9 @@
...
+SOURCES_CACHE: str = join(MAKES_DIR, "makes", "sources")
```
Comment (dsalaza4) [inline src/cli/main/cli.py:79]:
```suggestion
SOURCES_CACHE: str = join(MAKES_DIR, "sources")
```

File: src/evaluator/modules/default.nix [hunk 7]
```diff
@@ -59,7 +59,7 @@
   options = {
     globalStateDir = lib.mkOption {
       type = lib.types.str;
-      default = "$HOME_IMPURE/.makes/state";
+      default = "$HOME_IMPURE/.cache/state";
```
Comment (dsalaza4) [inline src/evaluator/modules/default.nix:62]:
```suggestion
      default = "$HOME_IMPURE/.cache/makes/state";
```

**Review by dsalaza4 — CHANGES_REQUESTED** [review]

**Comment by dsalaza4** [comment]
Please resolve previous threads accordingly

**Comment by dsalaza4** [comment]
@sebas031811 

Calling the TUI with `m .` and then selection a job seems to be failing after installing your CLI version with `nix-env -if .`.

To reproduce:

1. Install your cli with `nix-env -if .`
2. Run any makes job from the TUI by running `m .` first and then selecting a job.

**Comment by dsalaza4** [comment]
> @sebas031811
> 
> Calling the TUI with `m .` and then selection a job seems to be failing after installing your CLI version with `nix-env -if .`.
> 
> To reproduce:
> 
> 1. Install your cli with `nix-env -if .`
> 2. Run any makes job from the TUI by running `m .` first and then selecting a job.

I think it has to do with having an unnecessary second level `cache` directory here: 

https://github.com/fluidattacks/makes/blob/3004ae797b2ed44806cb9d923d0c4b293dd93c37/src/cli/main/cli.py#L547

**Review by dsalaza4 — APPROVED** [review]
LGTM!

**Pull request merged**